common/src/main/scala/react/floatingui/syntax.scala

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2016-2022 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package react.floatingui.syntax
+
+import japgolly.scalajs.react.*
+import japgolly.scalajs.react.vdom.html_<^.*
+import lucuma.ui.syntax.all.given
+import react.floatingui.*
+
+import scala.scalajs.js
+
+// Unfortunately, we can't differentiate between the extension methods below and use
+// `withTooltip` for both because a `VdomTag` IS a `VdomNode`.
```

Comment:
One last question: I dislike having extra crud in the DOM, but is it worth the extra extension method and name to avoid having an extra `span` in the DOM for tooltip triggers?

Reply:
The thing with `span` is that we need a way to set a `Ref` and you can't do that on `VdomNode`, you need a `VdomTag`, having a `span` around is the easiest way.
The extension seems nice from an API point of view but I wonder if it may obscure the fact that we are wrapping and you may need to adjust your css

Reply:
Maybe a bbetter option is to make `Toolitp` take a `ReactFnProps/GenericFNComponont/..` or something like that and set the `Ref` there

Reply:
I understand the need for the `span` with the `VdomNode`. My question was about avoiding wrapping a `span` in another `span`. I'm doing that now, but it requires the extra method for use with `VdomNode`.

Good point about it affecting the CSS. I was more considering the idea that the `ToolTip` wrapper was sort of obscuring what the "main" elements in the html were. 🤔

Reply:
> Maybe a bbetter option is to make `Toolitp` take a `ReactFnProps/GenericFNComponont/..` or something like that and set the `Ref` there

Hmmm. Let me see if I can come up with something like that.

Reply:
I do not see any way forward with something like that. (Not that I couldn't be missing something.) So, maybe we could just keep the syntax for elements (VdomTag), and just require the user to explicitly wrap components in `span`s. This leaves `span`s when they aren't needed (for optional tooltips), but it does keep the dom consistent. I still think it is still a win wrt syntax.

Reply:
I did the above, as well as remove the "optional" functionality. This should make it more apparent what you get in the DOM.